core/tool/tools.py

```diff
@@ -1,186 +1,103 @@
+    """Load the pickle object from file_path."""
     with open(file_path, 'rb') as f:
         return pickle.load(f)
 
 
-def load_raw_image(path, group='/image'):
-    if path[-2:] == 'h5':
+def load_raw_image(path: str, group: str = '/image') -> Tuple:
+    """Load the raw image volume."""
+    file_extension = os.path.splitext(path)[1]
+    if file_extension == '.h5':
         with h5py.File(path, 'r') as f:
             images = (f[group][()].astype(np.float32) - 128) / 33
     else:
         images = ((skimage.io.imread(path)).astype(np.float32) - 128) / 33
     return images
 
 
-
-def resume_dict_load(dict_path, name, resume_obj):
-    if os.path.exists(dict_path + name + '.pkl'):
-        resume = load_obj(dict_path + name + '.pkl')
+def resume_dict_load(dict_path: str, name: str, resume_obj: int) -> Dict:
+    """Load dict to resume processing."""
+    dict_path = os.path.join(dict_path, name, '.pkl')
+    if os.path.exists(dict_path):
+        resume = load_obj(dict_path)
     else:
         resume = {'resume_seed': resume_obj}
     resume_obj = resume['resume_seed']
-    print('resume', resume_obj)
+    print('Resume', resume_obj)
     return resume
 
 
-def resume_re_segd_count_mask(path,shape):
-    if os.path.exists(path + 're_seged_count_mask.tif'):
-        re_seged_count_mask = skimage.io.imread(path + 're_seged_count_mask.tif')
+def resume_re_segd_count_mask(file_path: str, shape: Tuple) -> Tuple:
+    """Load mask file if it exists otherwise create a new one."""
+    file_path = os.path.join(file_path, 're_seged_count_mask.tif')
+    if os.path.exists(file_path):
+        re_seged_count_mask = skimage.io.imread(file_path)
     else:
         re_seged_count_mask = np.zeros(shape, dtype=np.uint8)
     return re_seged_count_mask
 
-def seeds_to_dict(seeds_path):
-    inf_seed_dict = {}
-    if os.path.exists(seeds_path):
-        with h5py.File(seeds_path, 'r') as segs:
-            seeds = segs['seeds'][()]
-            seed_id = 1
-            seeds = list(seeds)
-            for coord in seeds:
-                inf_seed_dict[seed_id] = coord
-                seed_id += 1
-    else:
-        print("seeds file is not exist")
-
-    return inf_seed_dict
-
-
-
-
-
-
-
 
+def load_seeds_from_file(seeds_file_path: str, manual_seed: bool) -> Dict:
+    """
+    Load seeds either from a .h5 (auto generated seeds)
+    or from a .txt (manually specified seeds) file.
+    """
+    seeds = {}
+    if os.path.exists(seeds_file_path):
+        if not manual_seed:
+            with h5py.File(seeds_file_path, 'r') as f:
+                seeds = f['seeds'][()]
+                seeds = list(seeds)
+                seeds = {i + 1: coord for i, coord in enumerate(seeds)}
+        else:
+            # Expects a .txt file of seed coordinates in a Dict format
+            # Each line as {id: [z, y, x]}
+            with open(seeds_file_path, 'r') as f:
+                contents = f.read()
+                seeds = ast.literal_eval(contents)
```

Comment:
Not really important, but using ast.literal_eval to parse simple data like this seems like a non-standard way to serialize things. I would have used one seed per line with id: z, y, x with "manual line parsing" or json / YAML.

Reply:
I'll add this as a part of a later change if we decide to go for this.